Tests/Functional/Mailer/Transport/SparkpostTransportTest.php

```diff
@@ -26,6 +26,7 @@ public function testEmailSendToContactSync(): void
     {
         $expectedResponses = [
             function ($method, $url, $options): MockResponse {
+                die('this does not seem to be called at all');
```

Comment:
The reason this code is not called is because we need to explicitly mention what service we need to inject for SparkpostTransportFactory,php. Ref: https://github.com/ts-navghane/sparkpost-plugin/commit/4d3c90b0083ba0e9a55f039877bfa6c974e86261#diff-672d3c640fb2c9c9851db599c909a96f322869f0b9684d63f36a3fe8b77bb14dR19-R26.

If we remove this, the MockHttpClient from config_test.php is override by the container when we autowire it.